src/view/dataElements/eventMergeId.jsx

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import "regenerator-runtime"; // needed for some of react-spectrum
+import React from "react";
+import Select from "@react/react-spectrum/Select";
+import FieldLabel from "@react/react-spectrum/FieldLabel";
+import Alert from "@react/react-spectrum/Alert";
+import "@react/react-spectrum/Form"; // needed for spectrum form styles
+import { v4 as uuid } from "uuid";
+import render from "../render";
+import WrappedField from "../components/wrappedField";
+import ExtensionView from "../components/extensionView";
+import getInstanceOptions from "../utils/getInstanceOptions";
+import "./eventMergeId.styl";
+
+const getInitialValues = ({ initInfo }) => {
+  const {
+    instanceName = initInfo.extensionSettings.instances[0].name,
+    cacheId = uuid()
+  } = initInfo.settings || {};
+
+  return {
+    instanceName,
+    cacheId
+  };
+};
+
+const getSettings = ({ values }) => {
+  return values;
+};
+
+const EventMergeId = () => {
+  return (
+    <ExtensionView
+      getInitialValues={getInitialValues}
+      getSettings={getSettings}
+      render={({ initInfo }) => {
+        return (
+          <div>
+            <Alert variant="info" header="Event Merge ID Caching">
+              This data element will provide an event merge ID. Regardless of
+              what you choose for the data element storage duration in Launch,
+              the value of this data element will remain the same until either
+              the visitor to your website leaves the current page or the event
+              merge ID is reset using the Reset Event Merge ID action.
+            </Alert>
+            <div>
+              <FieldLabel labelFor="instanceNameField" label="Instance" />
+              <div>
+                <WrappedField
+                  data-test-id="instanceNameField"
+                  id="instanceNameField"
+                  name="instanceName"
+                  component={Select}
+                  componentClassName="u-fieldLong"
+                  options={getInstanceOptions(initInfo)}
+                />
```

Comment:
Technically, from within the data element library module, it's possible for us to just use the first instance we find to generate an event merge ID, since a random ID will be generated regardless of which instance we use. We still ask for which instance the customer wants to use to generate the event merge ID though because there will be log messages in the console that say which instance the `createEventMergeId` command is being run within and it would be weird if a customer used one instance 99% of the time and then sees a console log saying that a different instance was used to generate an event merge ID.

Reply:
One way to get rid of the instance selector is to just generate the id inside the launch code, and not call the alloy function.  We could log the creation in the satellite logger.  The instance selector makes sense for sending an event, but not really for the eventMergeId.

Reply:
Yeah, that would work. I wouldn't want to have duplicate uuidv4 code (one in the extension and one in alloy), so I'll probably want to make the reactor-specific Alloy build use the uuidv4 from the extension. Launch extensions don't support importing npm packages yet either, so that might make things tricky. Let me dig deeper and see what I find.